server/src/app/repositories/Base.repository.ts

```diff
@@ -0,0 +1,75 @@
+import { IDatabase } from "pg-promise";
+import {
+  DbExtensions, ERRORS, ErrorUtils, IBaseRepositiory
+} from "../index";
+
+export class BaseRepository <T> implements IBaseRepositiory <T> {
```

Comment:
I'm not sure if this is a bit to much at this point because I'm not able to anticipate how often we really could use it. But we will see.